From 7b570231784f9074ee5c9e687a2ef3d581de43f5 Mon Sep 17 00:00:00 2001
From: Matthew Johnson <matthew.johnson@intel.com>
Date: Mon, 15 Aug 2016 15:57:28 +0000
Subject: [PATCH 129/129] remove clear ioapic

Remove call to clear_IO_APIC in early boot

In hyper-v we can expect the IOAPIC to be empty on boot. The call
to clear_IO_APIC is then not necessary or desirable because it can
cause significant boot delays in hyper-v. We were seeing up to 300 ms
delays with this call on 32-core systems.
---
 arch/x86/kernel/apic/io_apic.c | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/arch/x86/kernel/apic/io_apic.c b/arch/x86/kernel/apic/io_apic.c
index 347bb9f65737..e3b8d9ee01f4 100644
--- a/arch/x86/kernel/apic/io_apic.c
+++ b/arch/x86/kernel/apic/io_apic.c
@@ -1425,9 +1425,12 @@ void __init enable_IO_APIC(void)
 	}
 
 	/*
-	 * Do not trust the IO-APIC being empty at bootup
+	 * In hyper-v we can trust that IO_APIC will be cleared on boot. A call to
+	 * clear_IO_APIC is therefore not necessary or desirable since it can
+	 * cause significant delays in boot time.
+	 *
+	 * clear_IO_APIC();
 	 */
-	clear_IO_APIC();
 }
 
 void native_disable_io_apic(void)
-- 
2.13.2

